src/main/java/de/tum/in/www1/artemis/service/connectors/Lti10Service.java

```diff
@@ -0,0 +1,287 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.BasicResponseHandler;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.imsglobal.lti.launch.LtiOauthVerifier;
+import org.imsglobal.lti.launch.LtiVerificationException;
+import org.imsglobal.lti.launch.LtiVerificationResult;
+import org.imsglobal.lti.launch.LtiVerifier;
+import org.imsglobal.pox.IMSPOXRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.thymeleaf.util.StringUtils;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.LtiOutcomeUrlRepository;
+import de.tum.in.www1.artemis.repository.ResultRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.web.rest.dto.LtiLaunchRequestDTO;
+import oauth.signpost.exception.OAuthException;
+
+@Service
+public class Lti10Service {
+
+    private final Logger log = LoggerFactory.getLogger(Lti10Service.class);
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final LtiService ltiService;
+
+    private final LtiOutcomeUrlRepository ltiOutcomeUrlRepository;
+
+    private final ResultRepository resultRepository;
+
+    private final HttpClient client;
+
+    public Lti10Service(UserRepository userRepository, LtiOutcomeUrlRepository ltiOutcomeUrlRepository, ResultRepository resultRepository, CourseRepository courseRepository,
+            LtiService ltiService) {
+        this.userRepository = userRepository;
+        this.ltiOutcomeUrlRepository = ltiOutcomeUrlRepository;
+        this.resultRepository = resultRepository;
+        this.courseRepository = courseRepository;
+        this.ltiService = ltiService;
+        this.client = HttpClientBuilder.create().build();
+    }
+
+    /**
+     * Checks if an LTI request is correctly signed via OAuth with the secret
+     *
+     * @param request The request to check
+     * @param onlineCourseConfiguration The configuration containing the secret used to verify the request
+     * @return null if the request is valid, otherwise an error message which indicates the reason why the verification failed
+     */
+    public String verifyRequest(HttpServletRequest request, OnlineCourseConfiguration onlineCourseConfiguration) {
```

Comment:
How about returning `bool` and throwing exceptions failures?

Reply:
This wasn't added in this PR, I just moved it from LtiService to a more specific Lti10Service. I'd rather tackle this in another PR